### PR TITLE
feat: Evaluate agents with GenAI Model Eval

### PR DESCRIPTION
**Pull request by inardini**
# Description

This PR is about the new Gen AI Evaluation for agent evaluation. 

Thank you for opening a Pull Request!
Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [x] Follow the [`CONTRIBUTING` Guide](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/CONTRIBUTING.md).
- [x] You are listed as the author in your notebook or README file.
  - [x] Your account is listed in [`CODEOWNERS`](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/.github/CODEOWNERS) for the file(s).
- [x] Make your Pull Request title in the <https://www.conventionalcommits.org/> specification.
- [x] Ensure the tests and linter pass (Run `nox -s format` from the repository root to format).
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: gemini/evaluation/evaluating_crewai_agent.ipynb [hunk 1]
```diff
@@ -0,0 +1,1573 @@
...
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "JAPoU8Sm5E6e"
+      },
+      "source": [
+        "# Evaluating Agents - Evaluate a CrewAI agent with Vertex AI Gen AI Evaluation\n",
+        "\n",
+        "<table align=\"left\">\n",
+        "  <td style=\"text-align: center\">\n",
+        "    <a href=\"https://colab.research.google.com/github/GoogleCloudPlatform/generative-ai/blob/main//generative-ai/gemini/evaluation/evaluating_crewai_agent.ipynb\">\n",
```
[the remark this answers is not in the document]
Comment (inardini) [reply gemini/evaluation/evaluating_crewai_agent.ipynb]: done

File: gemini/evaluation/evaluating_crewai_agent.ipynb [hunk 2]
```diff
@@ -0,0 +1,1573 @@
...
+        "      <img width=\"32px\" src=\"https://lh3.googleusercontent.com/JmcxdQi-qOpctIvWKgPtrzZdJJK-J3sWE1RsfjZNwshCFgE_9fULcNpuXYTilIR2hjwN\" alt=\"Google Cloud Colab Enterprise logo\"><br> Open in Colab Enterprise\n",
+        "    </a>\n",
+        "  </td>\n",
+        "  <td style=\"text-align: center\">\n",
+        "    <a href=\"https://console.cloud.google.com/vertex-ai/workbench/deploy-notebook?download_url=https://raw.githubusercontent.com/GoogleCloudPlatform/generative-ai/main/gemini/evaluation/evaluating_crewai_agent.ipynb\">\n",
+        "      <img src=\"https://www.gstatic.com/images/branding/gcpiconscolors/vertexai/v1/32px.svg\" alt=\"Vertex AI logo\"><br> Open in Vertex AI Workbench\n",
+        "    </a>\n",
+        "  </td>\n",
+        "  <td style=\"text-align: center\">\n",
+        "    <a href=\"https://github.com/GoogleCloudPlatform/generative-ai/blob/main//generative-ai/gemini/evaluation/evaluating_crewai_agent.ipynb\">\n",
```
[the remark this answers is not in the document]
Comment (inardini) [reply gemini/evaluation/evaluating_crewai_agent.ipynb]: done

File: gemini/evaluation/evaluating_langgraph_agent.ipynb [hunk 3]
```diff
@@ -0,0 +1,1561 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "84f0f73a0f76"
+      },
+      "source": [
+        "| | |\n",
+        "|-|-|\n",
+        "| Authors | [Ivan Nardini](https://github.com/inardini) [Naveksha Sood](https://github.com/navekshasood)|"
```
[the remark this answers is not in the document]
Comment (inardini) [reply gemini/evaluation/evaluating_langgraph_agent.ipynb]: done

File: gemini/reasoning-engine/evaluating_crewai_agent_reasoning_engine_customized_template.ipynb [hunk 4]
```diff
@@ -0,0 +1,1696 @@
...
+        "    PROJECT_ID = str(os.environ.get(\"GOOGLE_CLOUD_PROJECT\"))\n",
+        "\n",
+        "LOCATION = os.environ.get(\"GOOGLE_CLOUD_REGION\", \"us-central1\")\n",
+        "\n",
+        "BUCKET_NAME = \"[your-bucket-name]\"  # @param {type: \"string\", placeholder: \"[your-bucket-name]\", isTemplate: true}\n",
+        "\n",
+        "if not BUCKET_NAME or BUCKET_NAME == \"[your-bucket-name]\":\n",
+        "    BUCKET_NAME = f\"{PROJECT_ID}-bucket\"\n",
+        "\n",
+        "BUCKET_URI = f\"gs://evaluate_agents\"\n",
```
[the remark this answers is not in the document]
Comment (inardini) [reply gemini/reasoning-engine/evaluating_crewai_agent_reasoning_engine_customized_template.ipynb]: done

File: gemini/reasoning-engine/evaluating_langchain_agent_reasoning_engine_prebuilt_template.ipynb [hunk 5]
```diff
@@ -0,0 +1,1529 @@
...
+        "\n",
+        "PROJECT_ID = \"[your-project-id]\"  # @param {type: \"string\", placeholder: \"[your-project-id]\", isTemplate: true}\n",
+        "\n",
+        "if not PROJECT_ID or PROJECT_ID == \"[your-project-id]\":\n",
+        "    PROJECT_ID = str(os.environ.get(\"GOOGLE_CLOUD_PROJECT\"))\n",
+        "\n",
+        "LOCATION = os.environ.get(\"GOOGLE_CLOUD_REGION\", \"us-central1\")\n",
+        "\n",
+        "BUCKET_NAME = \"[your-bucket-name]\"  # @param {type: \"string\", placeholder: \"[your-bucket-name]\", isTemplate: true}\n",
+        "BUCKET_URI = f\"gs://evaluate_agents\"\n",
```
[the remark this answers is not in the document]
Comment (inardini) [reply gemini/reasoning-engine/evaluating_langchain_agent_reasoning_engine_prebuilt_template.ipynb]: done

File: gemini/evaluation/evaluating_crewai_agent.ipynb [hunk 6]
```diff
@@ -0,0 +1,1573 @@
...
+        "    try:\n",
+        "        # Access tools_results directly from each agent\n",
+        "        for agent in crew.agents:\n",
+        "            if hasattr(agent, \"tools_results\"):\n",
+        "                for tool_result in agent.tools_results:\n",
+        "                    tool_info = {\n",
+        "                        \"tool_name\": tool_result.get(\"tool_name\", \"\"),\n",
+        "                        \"tool_input\": tool_result.get(\"tool_args\", {}),\n",
+        "                    }\n",
+        "                    final_output[\"predicted_trajectory\"].append(tool_info)\n",
```
[the remark this answers is not in the document]
Comment (inardini) [reply gemini/evaluation/evaluating_crewai_agent.ipynb]: done

**Review by holtskinner — APPROVED** [review]

**Pull request merged**